src/main/webapp/app/shared/components/authorization-checker/authorization-checker.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { IRootState } from 'app/shared/reducers';
+import CCRole from 'app/shared/model/enum/cc-role.enum';
+import EventRole from 'app/shared/model/enum/event-role.enum';
+
+interface IAuthorizationCheckerOwnProps {
+  ccRole?: CCRole;
+  eventRole?: EventRole;
+  eventId?: number;
+  fallbackEl?: React.ReactElement;
+}
+
+interface IAuthorizationCheckerProps extends IAuthorizationCheckerOwnProps, StateProps {}
+
+class AuthorizationChecker extends React.Component<IAuthorizationCheckerProps, {}> {
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    const {
+      children,
+      ccRole,
+      eventRole,
+      fallbackEl,
+      eventId,
+      isAuthenticated,
+      isCurrentCCHead,
+      isCurrentAdministrator,
+      isEventHead,
+      isEventCrew
+    } = this.props;
+    const fallbackElement: React.ReactElement | null = fallbackEl ? fallbackEl : null;
+    if (!isAuthenticated) return fallbackElement;
+    let hasPassCCRoleChecking = false;
+    let hasPassEventRoleChecking = false;
+    const hasCCRole = typeof ccRole !== 'undefined';
+    const hasEventRole = typeof eventRole !== 'undefined';
+    const hasEventId = typeof eventId !== 'undefined';
+    if (hasCCRole) {
+      const isOnlyCCHeadCanView = ccRole === CCRole.HEAD;
+      const isOnlyCCAdminCanView = ccRole === CCRole.ADMIN;
+      const hasPassCCHeadCheck = isOnlyCCHeadCanView && isCurrentCCHead;
+      const hasPassCCAdminCheck = isOnlyCCAdminCanView && isCurrentAdministrator;
+      hasPassCCRoleChecking = hasPassCCHeadCheck || hasPassCCAdminCheck;
+    }
+    if (hasEventRole && hasEventId) {
+      const isOnlyEventHeadCanView = eventRole === EventRole.HEAD;
+      const isOnlyEventCrewCanView = eventRole === EventRole.CREW;
+      const hasPassEventHeadCheck = isOnlyEventHeadCanView && isEventHead;
+      const hasPassEventCrewCheck = isOnlyEventCrewCanView && isEventCrew;
+      hasPassEventRoleChecking = hasPassEventHeadCheck || hasPassEventCrewCheck;
+    }
+    const canRender = hasPassCCRoleChecking || hasPassEventRoleChecking;
+    if (!canRender) return fallbackElement;
+    return children;
```

Comment:
Maybe can simplified into return canRender? fallbackElement: children?

Reply:
fixed in c46b7ec97d7da4f9b39f77bc0fa237fe73f54a0e